### PR TITLE
Migrate Terminal off os.fork in favor of Popen

### LINKED ISSUE #60504: [BUG] Terminal class hangs when run with threads
**Description**

While debugging #59903 we discovered that `salt.utils.vt.Terminal` will sometimes hang before running the command. This problem happens when running in a multi-threaded environment. We tracked the hang down to some book keeping done by python's `os.fork` method.e

Migrate `salt.utils.vt.Terminal` use `subprocess.Popen` instead of making a call to `os.fork`. `subprocess.Popen` forks at the C level and avoids the same bookkeeping done by os.fork

### PR DESCRIPTION
**Pull request by dwoz**
### What does this PR do?

Changes `salt.utils.vt.Terminal` to use `subprocess.Popen` instead of calling `os.fork`. This avoids python's built in book keeping which happens on `os.fork` calls and can cause dead locks when used in multi-threaded environments.

### What issues does this PR fix or reference?

Fixes: #60504

### Previous Behavior
Remove this section if not relevant

### New Behavior
Remove this section if not relevant

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### CODE REVIEW

File: salt/utils/vt.py [hunk 1]
```diff
@@ -788,38 +666,10 @@ def getwinsize(self):
...
-            # Note that this fix is a hack.
-            TIOCSWINSZ = getattr(termios, "TIOCSWINSZ", -2146929561)
-            if TIOCSWINSZ == 2148037735:
-                # Same bits, but with sign.
-                TIOCSWINSZ = -2146929561
-            # Note, assume ws_xpixel and ws_ypixel are zero.
-            packed = struct.pack(b"HHHH", rows, cols, 0, 0)
-            fcntl.ioctl(self.child_fd, TIOCSWINSZ, packed)
+        def setwinsize(self, child, rows=80, cols=80):
+            setwinsize(self.child_fd, rows, cols)
```
Comment (s0undt3ch) [inline salt/utils/vt.py:672]: There's no scoping issues with the functions being named the same?

**Review by s0undt3ch — DISMISSED** [review]

File: salt/utils/vt.py [hunk 2]
```diff
@@ -61,31 +63,36 @@ class TerminalException(Exception):
...
 
-# ----- Cleanup Running Instances ------------------------------------------->
-# This lists holds Terminal instances for which the underlying process had
-# not exited at the time its __del__ method got called: those processes are
-# wait()ed for synchronously from _cleanup() when a new Terminal object is
-# created, to avoid zombie processes.
-_ACTIVE = []
-
-
-def _cleanup():
```
Comment (waynew) [inline salt/utils/vt.py:72]: This function was removed, but https://github.com/saltstack/salt/blob/ff31d08079d2100f49c6471461bc22724bf6c392/salt/utils/vt.py#L129 relies on it

**Review by waynew — DISMISSED** [review]
Found a bug in here. I'm not familiar enough with programmatic access to virtual terminals to know if there are any other issues, but the tests did at least pass (the couple that ran) after fixing the `_cleanup` bit

File: salt/utils/vt.py (hunk 1 above)
Comment (dwoz) [reply salt/utils/vt.py:669]: No, it's a simple closure. :)

File: salt/utils/vt.py (hunk 2 above)
Comment (dwoz) [reply salt/utils/vt.py:72]: fixed

File: tests/unit/utils/test_vt.py [hunk 3]
```diff
@@ -64,26 +64,37 @@ def wrapper_fixStdOutErrFileNoIfNeeded(*args, **kwargs):
...
-        True,
-        "Disabled until we can figure out why this fails when whole test suite runs.",
+        salt.utils.platform.is_windows(),
+        "Skip on Windows because this feature is not supported",
     )
     def test_vt_size(self):
         """Confirm that the terminal size is being set"""
-        if not sys.stdin.isatty():
-            self.skipTest("Not attached to a TTY. The test would fail.")
+        #        if not sys.stdin.isatty():
```
Comment (s0undt3ch) [inline tests/unit/utils/test_vt.py:72]: These comments are no longer useful right?

Comment (dwoz) [reply tests/unit/utils/test_vt.py]: true

**Review by s0undt3ch — APPROVED** [review]

**Comment by dwoz** [comment]
These changes went into the freeze and will get merged forward with 3002.2

**Pull request closed without merging**